src/MarketingPageAcceptanceTests.TestData/Solutions/ClientApplicationStringBuilder.cs

```diff
@@ -8,51 +11,45 @@ public static class ClientApplicationStringBuilder
 
         public static string GetClientAppString(string ignoredSection = null, string clientApplicationTypes = "Browser based")
         {
-            string clientAppString = string.Empty;
+            List<string> clientAppString = new List<string>();
+
+            string finishedString = string.Empty;
             if (string.IsNullOrEmpty(ignoredSection))
             {
                 if(clientApplicationTypes.Contains(browserBased))
                 {
-                    clientAppString += ClientApplicationStrings.BrowserBased;
+                    clientAppString.Add(ClientApplicationStrings.BrowserBased);
                 }
 
                 if (clientApplicationTypes.Contains(nativeMobile))
                 {
-                    if (!string.IsNullOrEmpty(clientAppString))
-                    {
-                        clientAppString += ",";
-                    }
-                    clientAppString += ClientApplicationStrings.NativeMobile;
+                    clientAppString.Add(ClientApplicationStrings.NativeMobile);
                 }
 
                 if (clientApplicationTypes.Contains(nativeDesktop))
                 {
-                    if (!string.IsNullOrEmpty(clientAppString))
-                    {
-                        clientAppString += ",";
-                    }
-                    clientAppString += ClientApplicationStrings.NativeDesktop;
+                    clientAppString.Add(ClientApplicationStrings.NativeDesktop);                    
                 }
 
-                return buildClientApplicationString(clientApplicationTypes, clientAppString);
+                finishedString = buildClientApplicationString(clientApplicationTypes, string.Join(',', clientAppString));
```

Comment:
doesn't the ignoredSection being null and carrying on cause an error when it reaches the next block of IFs?